### PR TITLE
Fix List with a simple grouped structure

### PR DESCRIPTION
**Pull request by KsyushaM**
After #6417

### CODE REVIEW

**Review by churkin — CHANGES_REQUESTED** [review]
Can we add test for this?

File: js/ui/shared/grouped_data_converter_mixin.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+const isObject = require("../../core/utils/type").isObject;
```
Comment (DokaRus) [inline js/ui/shared/grouped_data_converter_mixin.js]:
```suggestion
import { isObject } from "../../core/utils/type";
```

File: js/ui/shared/grouped_data_converter_mixin.js [hunk 2]
```diff
@@ -28,6 +35,7 @@ export default {
                     type: "array",
                     data: dataSource
                 },
+                searchExpr: "text",
```
Comment (lukyanovas) [inline js/ui/shared/grouped_data_converter_mixin.js]: use searchExpr: "text", only for non-objects

Comment (KsyushaM) [reply js/ui/shared/grouped_data_converter_mixin.js]: Sure

**Comment by KsyushaM** [comment]
Ready to review

**Review by DokaRus — DISMISSED** [review]

**Review by lukyanovas — DISMISSED** [review]

**Review by DokaRus — APPROVED** [review]

**Review by lukyanovas — APPROVED** [review]

**Review by churkin — APPROVED** [review]

**Pull request merged**